openedx/features/pakx/lms/pakx_admin_app/views.py

```diff
@@ -21,21 +35,49 @@ class UserProfileViewSet(viewsets.ModelViewSet):
     OrderingFilter.ordering_fields = ('id', 'name', 'email', 'employee_id')
     ordering = ['-id']
 
+    def create(self, request, *args, **kwargs):
+        if self.create_request_is_valid(request):
+            return self.create_new_user(request)
+
+        return Response({
+            "username": request.data.get('username', "REQUIRED"),
+            "email": request.data.get('email', "REQUIRED"),
+            "employee_id": request.data.get('employee_id', "REQUIRED"),
+            "language": request.data.get('language', "REQUIRED")},
+            status=status.HTTP_400_BAD_REQUEST
+        )
+
+    def partial_update(self, request, *args, **kwargs):
+        user = self.get_object()
+        self.specify_user_role(user, request.data.pop("role", LEARNER))
+        user.profile.employee_id = request.data.pop('employee_id', user.profile.employee_id)
+        user.profile.language = request.data.pop('language', user.profile.language)
+        serializer = self.get_serializer(user, data=request.data, partial=True)
+        if serializer.is_valid():
+            user.save(update_fields=('is_superuser', 'is_staff'))
+            user.profile.save(update_fields=('employee_id', 'language'))
+            serializer.save()
+            return Response(status=status.HTTP_200_OK)
+        return Response(status=status.HTTP_400_BAD_REQUEST)
```

Comment:
use serializer for partial update https://www.django-rest-framework.org/api-guide/serializers/#partial-updates